### DISCUSSION TITLE
Support for non-standards compliant GraphQL APIs

### DISCUSSION THREAD

**Discussion by poslogix** [graphql-dotnet/graphql-dotnet#3391] [Q&A]
Hi
I hope it's acceptable to ask this kind of question here.

I'm currently trying to integrate with a GraphQL API that was written before the standards.

They are returning an additional property 'validation' in the error response e.g.:
`{
  "errors": [
    {
      "message": "validation",
      "locations": [
        {
          "line": 2,
          "column": 29
        }
      ],
      "validation": {
        "currency": [
          "This field is required"
        ]
      },
      "customer_facing": false
    }
  ],
  "data": {
    "createBalanceAdjustment": null
  }
}`

I've had a good look around the docs but can't seem to find a way to get access to anything under 'validation', or simply get access to the raw response JSON.

Any help would be greatly appreciated.

Thanks
Dan

**Comment by Shane32** [comment]
Is this a question for the `GraphQL.Client` nuget package?  Or how to write a server that returns information in the described format?

**Reply by poslogix** [reply]
Hi 
Thanks for the quick response. 
I'm attempting to use the GraphQL.Client nuget package to talk with someone else's API. 
It's this API that that's returning the non-standard 'validation' property.

**Reply by Shane32** [reply]
This repo is for the server project; sorry I don't use the client project.  I'd suggest opening an issue here: https://github.com/graphql-dotnet/graphql-client/issues

@rose-a perhaps you can answer this

**Reply by poslogix** [reply]
Thanks Shane, I've created a new post over there. Please feel free to delete this one.
Dan

**Reply by Shane32** [reply]
ref: https://github.com/graphql-dotnet/graphql-client/issues/485
